APIMatic.Core/Pagination/Paginator.cs

```diff
@@ -0,0 +1,113 @@
+﻿using APIMatic.Core.Request;
+using APIMatic.Core.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using APIMatic.Core.Pagination.Strategies;
+using System.Threading;
+
+namespace APIMatic.Core.Pagination
+{
+    /// <summary>
+    /// Provides synchronous iteration over a paginated data source by fetching items across multiple pages.
+    /// </summary>
+    /// <typeparam name="TItem">
+    /// The type of individual items to be iterated over.
+    /// </typeparam>
+    /// <typeparam name="TPageMetadata">
+    /// The type of metadata associated with each page of results, such as pagination state or response headers.
+    /// </typeparam>
+    public class Paginator<TItem, TPageMetadata> : IEnumerable<TItem>
```

Comment:
Why is this generic type named `TPageMetadata` and not `TPage`?
Same goes for this same name in `AsyncPaginator`